sdks/java/core/src/main/java/org/apache/beam/sdk/testing/PAssert.java

```diff
@@ -950,4 +1073,114 @@ public Void apply(Iterable<T> actual) {
       return new AssertContainsInAnyOrder<T>(expectedElements);
     }
   }
+
+  ////////////////////////////////////////////////////////////////////////////////////////////////
+
+  /**
+   * A strategy for filtering and rewindowing the actual and dummy {@link PCollection PCollections}
+   * within a {@link PAssert}.
+   */
+  private interface AssertionWindowingStrategy {
```

Comment:
Naming is a bit confusing, since it isn't a `WindowingStrategy` in the sense that exists in code. But just a nit.

More broadly, it isn't obvious what the realm of correct implementations might be.